project_recalculate/models/project_task.py

```diff
@@ -57,14 +60,17 @@ def _dates_onchange(self, vals):
         start = from_string(date_start)
         end = from_string(date_end)
         resource, calendar = self._resource_calendar_select()
+
+        if not start.tzinfo:
```

Comment:
For saving a lot of code lines, call directly in all places to _resource_timezone, and there, you check:
```python
if dt.tzinfo:
    return dt
```